src/core/mesh/qgstopologicalmesh.h

```diff
@@ -0,0 +1,277 @@
+     */
+    QgsMeshEditingError canFacesBeRemoved( const QList<int> facesIndexes );
+
+    /**
+     *  Removes faces with index in \a faceIndexes.
+     *  The method returns a instance of the class QgsTopologicalMesh::Change that can be used to reverse or reapply the operation.
+     */
+    Changes removeFaces( const QList<int> facesIndexes );
+
+    /**
+     *  Add a \a vertex in the face with index \a faceIndex. The including face is removed and new faces surrounding the added vertex are added.
+     *  The method returns a instance of the class QgsTopologicalMesh::Change that can be used to reverse or reapply the operation.
+     */
+    Changes addVertexInface( int faceIndex, const QgsMeshVertex &vertex );
+
+    /**
+     *  Add a free \a vertex in the face, that is a vertex tha tis not included or linked with any faces.
+     *  The method returns a instance of the class QgsTopologicalMesh::Change that can be used to reverse or reapply the operation.
+     */
+    Changes addFreeVertex( const QgsMeshVertex &vertex );
+
+    /**
+     * Removes the vertex with index \a vertexIndex.
+     * If the vertex in linked with faces, the operation leads also to remove the faces. In this case, if the flag \a fillHole is set to TRUE,
+     * the hole is filled by a triangulation.
+     * The method returns a instance of the class QgsTopologicalMesh::Change that can be used to reverse or reapply the operation.
+     */
+    Changes removeVertex( int vertexIndex, bool fillHole = false );
+
+    //! Applies the changes
+    void applyChanges( const Changes &changes );
+
+    //! Reverses the changes
+    void reverseChanges( const Changes &changes );
+
+    //! Check the topology of the face and set it counter clock wise if necessary
+    static QgsMeshEditingError counterClockWiseFaces( QgsMeshFace &face, QgsMesh *mesh );
+
+    //! Reindexes faces and vertices, after this operation, topological can't be edited anymore
+    void reindex();
+
+  private:
+
+    //! Creates topological faces from mesh faces
+    TopologicalFaces  createTopologicalFaces(
+      const QVector<QgsMeshFace> &faces,
+      QgsMeshEditingError &error,
+      bool allowUniqueSharedVertex,
+      bool writeInVertices );
+
+    QgsMeshVertexCirculator vertexCirculator( int vertexIndex ) const;
+    static bool facesCanBeJoinedWithCommonIndex( const QgsMeshFace &face1, const QgsMeshFace &face2, int commonIndex );
+
+    QSet<int> concernedFacesBy( const QList<int> faceIndex ) const;
+
+    //Attributes
+    QgsMesh *mMesh = nullptr;
+    QVector<int> mVertexToface;
+    QVector<FaceNeighbors> mFacesNeighborhood;
+
+    friend class QgsMeshVertexCirculator;
+
+};
+
+/**
+ * \ingroup core
+ *
+ * \brief  Convenient class that turn around a vertex and provide information about faces and vertices
+ *
+ * \since QGIS 3.22
+ */
+class QgsMeshVertexCirculator
```

Comment:
move to qgsmeshvertexcirculator.cpp[h].

Also it is not clear from the class name and description what the class does. is it described in qep somewhere? maybe topologicalMeshTraveller? not sure

Reply:
not sure for the move, it is a quite particular class that is completely link to the topological mesh.
class inspired by: https://doc.cgal.org/latest/Circulator/index.html#circulatorsCirculators

Reply:
I prefer to have any more substantial class in separate file, but it is not requirement, just my preference/wish